Transactions.md

```diff
@@ -1,182 +1,292 @@
+
+# Fees
+
+### Fee Calculation
+
+All fee calculations for all transactions is based on the fee rate specified in the offer message and the *expected weight* of the transaction in question.
+
+The actual and expected weights vary for several reasons:
+
+* Bitcoin uses DER-encoded signatures, which vary in size.
+* Bitcoin also uses variable-length integers, so a large number of outputs will take 3 bytes to encode rather than 1.
+* The offerer output may be below the dust limit.
+* The accepter output may be below the dust limit.
+
+Thus, a simplified formula for *expected weight* is used, which assumes:
+
+* Signatures are 72 bytes long.
+* There are a small number of outputs (thus 1 byte to count them).
+* There are always both an offerer output and an accepter output.
+
+This yields the following *expected weights* (details of the computation below):
+
+```
+Funding Transaction weight:    286 + 4 * total_change_length + 272 * num_inputs
+CET/Refund Transaction weight: 500 + 4 * total_output_length
+```
+
+### Fee Payment
+
+All fees are paid for in the [Funding Transaction](#funding-transaction) so that the funding output's value is inflated so that the outputs of CETs and the refund transaction are the exact amounts specified in the offer message's contract information (or total collateral specified in the offer and accept messages for the refund transaction) with no fees subtracted.
+
+All fees are currently paid evenly between the two parties, though this will change in a future version.
+
+Note that if an outcome occurs in which one party's output is below the dust limit of `1000 satoshis`, then the resulting fee rate will be larger than *expected*.
+
+## Expected Weight of the Funding Transaction
+
+The *expected weight* of a funding transaction is calculated as follows:
+
+```
+p2wpkh: 22 bytes
+	- OP_0: 1 byte
+	- OP_DATA: 1 byte (public_key_HASH160 length)
+	- public_key_HASH160: 20 bytes
+
+p2wsh: 34 bytes
+	- OP_0: 1 byte
+	- OP_DATA: 1 byte (witness_script_SHA256 length)
+	- witness_script_SHA256: 32 bytes
+
+change_output: 9 + change_spk_script length bytes
+	- value: 8 bytes
+	- var_int: 1 byte (pk_script length)
+	- pk_script: change_spk_script length
+
+funding_output: 43 bytes
+	- value: 8 bytes
+	- var_int: 1 byte (pk_script length)
+	- pk_script (p2wsh): 34 bytes
+
+funding_input: 41 bytes
+	- previous_out_point: 36 bytes
+		- hash: 32 bytes
+		- index: 4 bytes
+	- var_int: 1 byte (script_sig length)
+	- script_sig: 0 bytes
+	- witness <----	"witness" is used instead of "script_sig" for
+ 			transaction validation; however, "witness" is stored
+ 			separately, and the cost for its size is smaller. So,
+ 		    the calculation of ordinary data is separated
+ 			from the witness data.
+	- sequence: 4 bytes
+
+witness_header: 2 bytes
+	- flag: 1 byte
+	- marker: 1 byte
+
+witness: 108 bytes
```

Comment:
so if I understand this correctly, we are limiting the funding of our transaction to be a p2sh(p2wpkh) or p2wpkh tx? It seems that are weight calculations are implying this any way.

Reply:
https://github.com/discreetlogcontracts/dlcspecs/issues/52